### PR TITLE
Update global files for sunbeam charms

### PR DESCRIPTION
**Pull request by hemanthnakkina**
Create tox, requirements, test-requirements for sunbeam charms
Update scripts to handle new charm structure ops-unknown

### CODE REVIEW

**Review by ajkavanagh — CHANGES_REQUESTED** [review]
I think there is an opportunity to introduce an `upper-contraints.txt` file to the global config that lists the upper bounds of python modules, and then using the '-c' on 'pip' (and related friends) when installing modules/building charms.  See [here](https://pip.pypa.io/en/stable/user_guide/#constraints-files) for more details.

This is something I'd like to roll out across all the charms by the end of the dev cycle and this seems like a good place to start.

Also, we definitely want to re-pin pip to < 23 for this round; we pinned to <20.3 previously because towards the end of a dev cycle (and close to release), pip release 20.3.2 that broke everything as we don't have good dependencies sorted out.  However, there's a golden opportunity to resolve this at this stage in the cycle, so it might be useful to sort it now.

File: _update-tox-files [hunk 1]
```diff
@@ -43,12 +46,23 @@ for charm in $charms; do
...
-        # Death to py27
-        sed -e "s#python-charm-jobs#python35-charm-jobs#g" -i charms/$charm/.zuul.yaml
+        case $charm_type in
+            classic-zaza|source-zaza)
+                # Death to py27
+                sed -e "s#python-charm-jobs#python35-charm-jobs#g" -i charms/$charm/.zuul.yaml
+                ;;
+            ops-unknown)
+                ;;
+        esac
```
[the remark this answers is not in the document]
Comment (hemanthnakkina) [reply _update-tox-files]: Done

File: global/README.md [hunk 2]
```diff
@@ -1,7 +1,8 @@
-There are 2 basic OpenStack Charm structure:
+There are 3 basic OpenStack Charm structure:
 
 1. Classic (non-reactive) + Zaza functional tests
 2. Source (reactive) + Zaza functional tests
+3. Ops (Operator framework)
```
[the remark this answers is not in the document]
Comment (hemanthnakkina) [reply global/README.md:5]: I am not sure about this and left with 3 types of charms for now. We can change names and add new type once the test framework is decided and new ops machine charms are created.

File: global/ops-unknown/tox.ini [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+#   https://github.com/pypa/pip/issues/9187
+# * Pinning dependencies requires tox >= 3.2.0, see
+#   https://tox.readthedocs.io/en/latest/config.html#conf-requires
+# * It is also necessary to pin virtualenv as a newer virtualenv would still
+#   lead to fetching the latest pip in the func* tox targets, see
+#   https://stackoverflow.com/a/38133283
+requires =
+  pip < 20.3
+  virtualenv < 20.0
+  setuptools<50.0.0
```
[the remark this answers is not in the document]
Comment (hemanthnakkina) [reply global/ops-unknown/tox.ini]: For now, removed all pinning to see if everything works fine. As of now there are no issues (atleast with pep8, py3 tests). Can we update these constraints as and when we see issues?

File: global/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ajkavanagh) [reply global/README.md:5]: We do already have at least one ops framework machine charm: https://github.com/openstack/charm-manila-netapp

So it definitely exists. I guess my question is "are they similar enough to be treated as one category, or should they be treated separately" from a tox, charmcraft, etc. perspective?

[the remark this answers is not in the document]
Comment (gnuoy) [reply global/README.md:5]: ceph-iscsi, ceph-dashboard, openstack-loadbalancer are all ops machine charms. At the moment the ops machine charms and the ops k8s charms are different categories. They use different libraries and have different structure.

File: _update-tox-files [hunk 4]
```diff
@@ -22,6 +22,9 @@ for charm in $charms; do
                 cp -fvp global/$charm_type/src/tox.ini charms/$charm/src/tox.ini
                 cp -fvp global/$charm_type/rename.sh charms/$charm/rename.sh
                 ;;
+            ops-unknown)
```
Comment (gnuoy) [inline _update-tox-files:25]: I think we should fix whatever is generating the category name *ops-unknown* (it's probably the script `what-is`). I think we need ops-k8s and ops-machine or something similar

**Review by ajkavanagh — APPROVED** [review]

**Pull request merged**